In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

### Loading data from CSV file into data frame.

In [3]:
appsdata = pd.read_csv('apps.csv')
appsdata.head()

Unnamed: 0                                                App  \
0           0     Photo Editor & Candy Camera & Grid & ScrapBook   
1           1                                Coloring book moana   
2           2  U Launcher Lite – FREE Live Cool Themes, Hide ...   
3           3                              Sketch - Draw & Paint   
4           4              Pixel Draw - Number Art Coloring Book   

         Category  Rating  Reviews  Size     Installs  Type Price  \
0  ART_AND_DESIGN     4.1      159  19.0      10,000+  Free     0   
1  ART_AND_DESIGN     3.9      967  14.0     500,000+  Free     0   
2  ART_AND_DESIGN     4.7    87510   8.7   5,000,000+  Free     0   
3  ART_AND_DESIGN     4.5   215644  25.0  50,000,000+  Free     0   
4  ART_AND_DESIGN     4.3      967   2.8     100,000+  Free     0   

  Content Rating                     Genres      Last Updated  \
0       Everyone               Art & Design   January 7, 2018   
1       Everyone  Art & Design;Pretend Play  January 15, 2018   
2       Everyone               Art & Design    August 1, 2018   
3           Teen               Art & Design      June 8, 2018   
4       Everyone    Art & Design;Creativity     June 20, 2018   

          Current Ver   Android Ver  
0               1.0.0  4.0.3 and up  
1               2.0.0  4.0.3 and up  
2               1.2.4  4.0.3 and up  
3  Varies with device    4.2 and up  
4                 1.1    4.4 and up

In [4]:
appsdata.shape

(9659, 14)

### Columns of dataset.

In [9]:
appsdata.columns

Index(['Unnamed: 0', 'App', 'Category', 'Rating', 'Reviews', 'Size',
       'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated',
       'Current Ver', 'Android Ver'],
      dtype='object')

In [11]:
appsdata.dtypes

Unnamed: 0          int64
App                object
Category           object
Rating            float64
Reviews             int64
Size              float64
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

### Primary observations of dataset.

- The dataset has 9,659 rows and 14 columns.
- Following columns are there in the dataset.
    - Unnamed: 0
    - App
    - Category
    - Rating
    - Reviews
    - Size
    - Installs
    - Type
    - Price
    - Content Rating
    - Genres
    - Last Updated
    - Current Ver
    - Android Ver
- Therea are 4 numeric columns in the dataset.
- By looking at the data, there is still a scope of changing to appropriate types for some columns.

### Null value analysis.

Let's look at null values in the dataset.

In [16]:
(appsdata.isnull().sum() / len(appsdata)) * 100

Unnamed: 0         0.000000
App                0.000000
Category           0.000000
Rating            15.146495
Reviews            0.000000
Size              12.703178
Installs           0.000000
Type               0.000000
Price              0.000000
Content Rating     0.000000
Genres             0.000000
Last Updated       0.000000
Current Ver        0.082824
Android Ver        0.020706
dtype: float64

### Observation

- Rating and Size columns have 15% and 12% null values respectively.